### PR TITLE
HTML API: Fix minor issues detecting funky comments and token boundaries.

### PR DESCRIPTION
**Pull request by dmsnell**
Fixes Core-60170

In an audit of funky comment syntax some issues were uncovered relating to misidentifying them as other comments. This patch attempts to fix the identified issues.

| HTML | Before | After |
|---|---|---|
| `</ >` | Funky Comment: ` ><img` | Funky Comment: ` `, Tag: `IMG` |

It was also discovered that some of the token boundaries are not properly recorded. While this hasn't caused a problem with the public interface or calling code, it would start to pose problems when continuing the reliance on the bookmarks to perform inner and outer markup changes.

Follow-up to [60428].

### CODE REVIEW

**Comment by dmsnell** [comment]
cc: @cbravobernal @darerodz I've removed one of the Interactivity API tests because it was accidentally passing when it shouldn't have been. it was testing whether the Server Directive Processor bails on unbalanced content, except it was testing with a `</ >` which isn't a tag closer but rather a funky comment. That particular funky comment wasn't properly detected before, and fixing it broke the test.

File: tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php
Comment (sirreal) [inline tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php]: Wrong version looks like:
```suggestion
	 * @since 6.6.0
```

File: tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php
Comment (sirreal) [inline tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php]: This descriptive text seems less obvious than the others, I don't know what it means.

**Review by sirreal — APPROVED** [review]

File: tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php
Comment (dmsnell) [reply tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php]: Fixed in d374d5774d68357220fa46f5f991ab8338f163d3

File: tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php
Comment (dmsnell) [reply tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php]: I updated the wording to note a question mark. It was intended to indicate "looks like markdown declaration state" but that didn't need to be.

File: tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php
Comment (gziolo) [inline tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php:592]: Nice refactoring 👍🏻

File: tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php
Comment (gziolo) [inline tests/phpunit/tests/html-api/wpHtmlTagProcessor-token-scanning.php:811]: That's a few interesting concepts included in this test case:

- translations
- bits
- bit with attributes

I know it only illustrates the ideas. We will need to remember to update these examples if some of the concepts will go in a different direction.

**Review by gziolo — APPROVED** [review]
This looks nice 👍🏻

**Comment by gziolo** [comment]
@dmsnell, I assume you would like to include this fix in 6.5.3 as you reopened the ticket, right?

**Comment by dmsnell** [comment]
> would like to include this fix in 6.5.3 as you reopened the ticket, right?

@gziolo this could have been a mistake on my part, but I was told something that led me to believe I need to reopen a ticket if I'm making a bug-fix for it. maybe I misunderstood. people wanted more of the work to be consolidated on fewer Trac tickets, so I was trying to do that. should I re-close the ticket?

**Comment by dmsnell** [comment]
Merged in [[58040]](https://core.trac.wordpress.org/changeset/58040)
https://github.com/wordpress/wordpress-develop/commit/10c3dd9cfb4ec5c4370e55590d41dd6eddbe49a4

**Comment by gziolo** [comment]
Well, I don't know. @aaronjorbin, if I am not mistaken, you are leading 6.5.x releases, can you help clarify the workflow that @dmsnell explained in https://github.com/WordPress/wordpress-develop/pull/6412#issuecomment-2074280668.

> this could have been a mistake on my part, but I was told something that led me to believe I need to reopen a ticket if I'm making a bug-fix for it. maybe I misunderstood. people wanted more of the work to be consolidated on fewer Trac tickets, so I was trying to do that. should I re-close the ticket?

**Comment by aaronjorbin** [comment]
In general, tickets on closed milestones shouldn't be reopened but the ticket should still be referenced on the follow up commit so that when someone is later looking at the feature, they can see the followup work. 

The [doc on commit messages](https://make.wordpress.org/core/handbook/best-practices/commit-messages/#ticket-references) talks a bit about how to do this reference, but [here is an example](https://core.trac.wordpress.org/changeset/57306) 

To have a change considered for backport, the process should be to create a ticket on trac and put it in the 6.5.3 milestone. Once there, it follows a workflow with the `fixed-major` keyword to signify that code has been committed to trunk, `dev-feedback` to indicate that a second committer should sign off on it and `dev-reviewed` to show that the second committer has approved it.

**Comment by dmsnell** [comment]
Thanks @aaronjorbin. I don't think this needs to be back-ported, as it's probably a fairly rare bug in practice. Next time I'll leave the ticket closed.

**Pull request closed without merging**
